guava/src/com/google/common/math/LongMath.java

```diff
@@ -805,7 +821,10 @@ public static long factorial(int n) {
    *
    * @throws IllegalArgumentException if {@code n < 0}, {@code k < 0}, or {@code k > n}
    */
-  public static long binomial(int n, int k) {
+  @SuppressWarnings(value = {"lowerbound:compound.assignment.type.incompatible",// the lowest n can be is 0 in this method
+          "upperbound:array.access.unsafe.high"// (1): Since k <= n, k is safely indexed
```

Comment:
Is this solvable by one of the open issues for the Less Than type system?

Reply:
If you're referring to [Issue 2029](https://github.com/typetools/checker-framework/issues/2029), then yes, I think if UBC infer annotation @LessThan("#1 + 1") as @LTEqLengthOf("arr"), this should be fixed. This issue is in my TODO list in agenda as well, I'd like to take on this if it's alright.

Reply:
That is a reasonable issue to work on soon. In the meantime, you should add a link to that issue to this suppressed warning so that we remember to unsuppress it when the issue is fixed.